target/linux/lantiq/files/arch/mips/boot/dts/lantiq/ar10_tplink_archer-d2.dts

```diff
@@ -0,0 +1,268 @@
+                        gpios = <&gpio 34 GPIO_ACTIVE_LOW>;
+                        linux,code = <KEY_RESTART>;
+                };
+
+                wifi {
+                        label = "wifi";
+                        gpios = <&gpio 9 GPIO_ACTIVE_HIGH>;
+                        linux,code = <KEY_RFKILL>;
+                        linux,input-type = <EV_SW>;
+                };
+
+                wps {
+                        label = "wps";
+                        gpios = <&gpio 0 GPIO_ACTIVE_LOW>;
+                        linux,code = <KEY_WPS_BUTTON>;
+                };
+        };
+
+        leds: leds {
+                compatible = "gpio-leds";
+
+                led_power: power {
+                        label = "power";
+                        gpios = <&gpio 6 GPIO_ACTIVE_LOW>;
+                        default-state = "keep";
+                };
+
+                led_dsl: dsl {
+                        label = "dsl";
+                        gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+                };
+
+                led_internet: internet {
+                        label = "internet";
+                        gpios = <&gpio 5 GPIO_ACTIVE_LOW>;
+                };
+
+                usb {
+                        label = "usb";
+                        gpios = <&gpio 26 GPIO_ACTIVE_LOW>;
+                        trigger-sources = <&ehci_port1>;
+                        linux,default-trigger = "usbport";
+                };
+
+                lan {
+                        label = "lan";
+                        gpios = <&gpio 8 GPIO_ACTIVE_LOW>;
+                };
+
+                led_wlan: wlan {
+                        label = "wlan";
+                        gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+                };
+
+                wps {
+                        label = "wps";
+                        gpios = <&gpio 27 GPIO_ACTIVE_LOW>;
+                };
+        };
+
+        usb_vbus: regulator-usb-vbus {
+		compatible = "regulator-fixed";
+
+		regulator-name = "USB_VBUS";
+
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+
+		gpio = <&gpio 25 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+};
+
+&spi {
+        status = "okay";
+
+        flash@0 {
+                compatible = "jedec,spi-nor";
+                reg = <1>;
+                spi-max-frequency = <25000000>;
+                m25p,fast-read;
+
+                partitions {
+                        compatible = "fixed-partitions";
+                        #address-cells = <1>;
+                        #size-cells = <1>;
+
+                        partition@0 {
+                                reg = <0x0 0x20000>;
+                                label = "u-boot";
+                                read-only;
+                        };
+
+                        partition@20000 {
+                                reg = <0x20000 0xf90000>;
+                                label = "firmware";
```

Comment:
Does firmware partition contains GPHY firmware?

Reply:
As far as I understand, this is a kernel+rootfs partition, which gets overwriten by OpenWRT. But by the looks of things (and binwalk output) OEM firmware does seem to contain the GPHY firmware here (https://static.tp-link.com/res/down/soft/Archer_D2(EU)_V1_160216.zip)